src/test/java/project/eepw/softwaretestingcrud/domain/post/data/PostServiceTest.java

```diff
@@ -0,0 +1,199 @@
+        verify(postRepository, times(1)).findById(nullId);
+    }
+
+    @Test
+    void shouldReturnAllPostsWhenGetAllPostsInvoked() {
+        //given
+        Post firstPost = makePost();
+        Post secondPost = makePost().toBuilder()
+                .id(2L)
+                .content("Another post content")
+                .user(makeUser())
+                .build();
+
+        List<Post> expectedListOfPosts = List.of(firstPost, secondPost);
+
+        when(postRepository.findAll()).thenReturn(expectedListOfPosts);
+
+        //when
+        Collection<Post> allPosts = postService.getAllPosts();
+
+        //then
+        verify(postRepository, times(1)).findAll();
+        assertThat(allPosts)
+                .usingRecursiveComparison()
+                .isEqualTo(expectedListOfPosts);
+    }
+
+    @Test
+    void shouldReturnEmptyCollectionWhenNoPostsInDatabase() {
+        //given
+        when(postRepository.findAll()).thenReturn(Collections.emptyList());
+
+        //when
+        Collection<Post> allPosts = postService.getAllPosts();
+
+        //then
+        verify(postRepository, times(1)).findAll();
+        assertThat(allPosts)
+                .usingRecursiveComparison()
+                .isEqualTo(Collections.emptyList());
+    }
+
+    @Test
+    void shouldReturnSetOfUserPostsWhenGivenExistingUserId() {
+        //given
+        Post firstPost = makePost();
+        Post secondPost = makePost().toBuilder()
+                .id(2L)
+                .content("Another post content")
+                .build();
+        User user = makeUser().toBuilder()
+                .posts(Set.of(firstPost, secondPost))
+                .build();
+        Long userId = user.getId();
+
+        Collection<Post> expectedListOfPosts = Set.of(firstPost, secondPost);
+        when(userService.getUserById(userId)).thenReturn(user);
+
+        //when
+        Collection<Post> fetchedPosts = postService.getAllUserPosts(userId);
+
+        //then
+        verify(userService, times(1)).getUserById(userId);
+        assertThat(fetchedPosts)
+                .usingRecursiveComparison()
+                .isEqualTo(expectedListOfPosts);
+    }
+
+    @Test
+    void shouldReturnEmptySetOfUserPostsWhenGivenIdOfExistingUserWithoutPosts() {
+        //given
+        User user = makeUser();
+        Long userId = user.getId();
+
+        when(userService.getUserById(userId)).thenReturn(user);
+
+        //when
+        Collection<Post> fetchedPosts = postService.getAllUserPosts(userId);
+
+        //then
+        verify(userService, times(1)).getUserById(userId);
+        assertThat(fetchedPosts)
+                .usingRecursiveComparison()
+                .isEqualTo(Set.of());
+    }
+
+    @Test
+    void shouldThrowExceptionWhenGivenNonExistingUserId() {
+        //given
+        Long userId = 2L;
+
+        when(userService.getUserById(userId)).thenThrow(new UserNotFoundException("User has not been found"));
```

Comment:
Idk, when stubbing any methods it is better practise to use any() matcher than only userId matching